app/portainer/home/EnvironmentList/KubeconfigButton.tsx

```diff
@@ -20,9 +60,38 @@ export function KubeconfigButton({ environments }: Props) {
   }
 
   return (
-    <Button onClick={handleClick}>
-      <i className="fas fa-download space-right" /> kubeconfig
-    </Button>
+    <div>
+      <Button onClick={handleClick}>
+        <i className="fas fa-download space-right" /> kubeconfig
+      </Button>
+      <DialogOverlay
```

Comment:
IMO this should be a spearate component

Reply:
I have moved all dialog logic to prompt.

Reply:
but why leave the diaslog overlay here. it's part of the prompt

Reply:
why not the overlay